pkg/ignore/dockerignore.go

```diff
@@ -0,0 +1,242 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package ignore
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+// DockerIgnoreFile defines the name of the ignore file
+const DockerIgnoreFile = ".dockerignore"
+
+var (
+	_            Ignorer = &defaultIgnorer{}
+	excludedDirs         = sets.NewString()
+	includesDirs         = sets.NewString()
+	directory    string
```

Comment:
I don't understand why this is global state vs. fields in the `dockerIgnore` implementation?